### PR TITLE
Avoid rerenders caused by unappropriate store selectors and `currentTime`

### PR DESCRIPTION
**Pull request by bragaru-i**
### Tasks:
- apply swallow for multiple store slices: https://github.com/pmndrs/zustand#selecting-multiple-state-slices
- store `currentTime` in a ref(get it from `timeupdate` event) and use in places where it do not need a refresh UI(ex: forward/rewind buttons)
- `ReactPlayer` use in a memo component and distribute it to MediaContainer(avoid MediaContainer to collect for ReactPlayer props, that can trigger its children) 
- `ProgressBar` - use value as: [0.01,0.02] instead of [0.000001, 0.000011]  - this will save rerebders

### Mentions
- Using multiple selectors without "shallow" - you will lose "atomic" part of the store(something similar to redux's mapStateToProps) and your selector will update as often as MediaStore does(~on every 100 ms)
zustand [docs](https://github.com/pmndrs/zustand#selecting-multiple-state-slices): 
> If you want to construct a single object with multiple state-picks inside, similar to redux's mapStateToProps, you can tell zustand that you want the object to be diffed shallowly by passing the shallow equality function.
```ts
// Array pick, re-renders the component when either state.nuts or state.honey change
const [nuts, honey] = useBearStore(
  (state) => [state.nuts, state.honey],
  shallow
)
```

We can go with deep comparison, but at the moment shallow does enough for our usage(we store primitive data, except `alarms`, that is an array of numbers, but is always replaced with a new one(no filtering, reducing and so over) ).

```ts
const treats = useBearStore(
  (state) => state.treats,
  (oldTreats, newTreats) => compare(oldTreats, newTreats)
)
```
- Some buttons, like `Fwd` or `Rwd` buttons, need `currentTime` for skipping current time, but they do not need this value for updating its own UI. So, as a solution, better to get `currentTime` from events, and store it in a ref
- also `usePipHook` uses `currentTime`, but do not uses it for own UI purposal

### CODE REVIEW

**Comment by bragaru-i** [comment]
Main branch vs PR:
The only thing that is now "blinking" - on mouse move (it's throttled for every second)

https://user-images.githubusercontent.com/50721739/202375836-c45224a6-851c-47da-9861-13b04614dc3a.mp4

File: src/hooks/use-play-pause-replay.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ interface UsePlayPauseReplayHook {
 
 export const usePlayPauseReplayHook = (): UsePlayPauseReplayHook => {
 	const [isFinished, setIsFinished] = useOneMSDelayedState(false);
-	const listener = useMediaStore(state => state.getListener());
+	const listener = useMediaStore(state => state.getListener)();
```
Comment (ronnyroeller) [inline src/hooks/use-play-pause-replay.ts:20]: Is this change on purpose?

**Review by ronnyroeller — APPROVED** [review]
LGTM (pls see one comment). Pls also describe in PR what this is solving (e.g. what causes the rerenders) and how it's solved (e.g. how does "shallow" improve things)?

File: src/hooks/use-play-pause-replay.ts (hunk 1 above)
Comment (bragaru-i) [reply src/hooks/use-play-pause-replay.ts:20]: yes. calling `getListener` inside of `useMediaStore` will always cause different objects(and rerenders).
so we should get the function's ref from store (`getListener`)
